hnclient-v2/src/main/java/ca/bc/gov/hlth/hnclientv2/handshakeserver/ConnectionHandler.java

```diff
@@ -0,0 +1,432 @@
+package ca.bc.gov.hlth.hnclientv2.handshakeserver;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.net.Socket;
+import java.nio.charset.StandardCharsets;
+import java.util.Scanner;
+import java.util.concurrent.Callable;
+
+import org.apache.camel.ProducerTemplate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ca.bc.gov.hlth.hnclientv2.error.ErrorBuilder;
+import ca.bc.gov.hlth.hnclientv2.error.MessageUtil;
+import ca.bc.gov.hlth.hncommon.util.LoggingUtil;
+import io.netty.util.internal.StringUtil;
+
+public class ConnectionHandler implements Callable<Void> {
```

Comment:
(Not part of this commit) Wondering if we have enough junit for business logic in this class or we need to create a story to add junits?